db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryComparisonOperatorNode.java

```diff
@@ -232,6 +232,17 @@ else if (! rightTypeId.isStringTypeId() && leftTypeId.isStringTypeId())
             rightPadBitDataConstantNode((BitConstantNode) getLeftOperand(), (ColumnReference) getRightOperand());
         }
 
+        // TODO: Enable this code to allow native spark joins involving REALs.
```

Comment:
What should be done with this TODO? Should it be activated or it's not needed anymore and can be removed?

Reply:
@ipraznik-splice I just added this comment and example code as a reminder to do something like this in the future so we trigger more native spark joins.  Non-native spark can be very slow, but sometimes spark returns different results than splice, maybe due to storing Floats with a different amount of precision than us, or hashing Floats different from Doubles.  Adding this cast would cause the containsUnsafeSQLRealComparison method to return false in join code (such as BroadcastJoinOperation), and we'd pick up native spark join.  The downside is, the cast can remove our ability to use a predicate for index access, so I don't want to add it without seeing if there is a better way to do this.  But I want to leave it in as a reminder of work to be done.